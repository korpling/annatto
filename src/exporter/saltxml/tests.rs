use graphannis::{update::GraphUpdate, AnnotationGraph};
use insta::assert_snapshot;
use tempfile::TempDir;

use super::*;
use crate::{util::example_generator, StepID};

#[test]
fn export_corpus_structure() {
    let mut updates = GraphUpdate::new();
    example_generator::create_corpus_structure_two_documents(&mut updates);

    // Add some meta data to the documents
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc1".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "description".to_string(),
            anno_value: "A demo corpus".to_string(),
        })
        .unwrap();
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc1".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "author".to_string(),
            anno_value: "unknown".to_string(),
        })
        .unwrap();
    updates
        .add_event(graphannis::update::UpdateEvent::AddNodeLabel {
            node_name: "root/doc2".to_string(),
            anno_ns: "test".to_string(),
            anno_name: "description".to_string(),
            anno_value: "Another demo corpus".to_string(),
        })
        .unwrap();

    let mut g = AnnotationGraph::with_default_graphstorages(true).unwrap();
    g.apply_update(&mut updates, |_msg| {}).unwrap();

    let exporter = ExportSaltXml {};
    let output_path = TempDir::new().unwrap();
    let corpus_dir = output_path.path().join("root");
    std::fs::create_dir(&corpus_dir).unwrap();

    let step_id = StepID {
        module_name: "export_saltxml".to_string(),
        path: Some(corpus_dir.clone()),
    };

    exporter
        .export_corpus(&g, &corpus_dir, step_id.clone(), None)
        .unwrap();

    // There should be a saltProject.salt file
    let p = corpus_dir.join("saltProject.salt");
    assert_eq!(true, p.is_file());
    let result = std::fs::read_to_string(p).unwrap();
    assert_snapshot!(result);
}
