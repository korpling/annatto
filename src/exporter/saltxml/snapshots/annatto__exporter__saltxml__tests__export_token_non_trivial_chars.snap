---
source: src/exporter/saltxml/tests.rs
expression: doc
---
<?xml version="1.0" encoding="UTF-8"?>
<sDocumentStructure:SDocumentGraph xmlns:sDocumentStructure="sDocumentStructure" xmlns:saltCore="saltCore" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:version="2.0">
  <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1"/>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tok4"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tok4"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tok3"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tok3"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tok2"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tok2"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tok1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tok1"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:SToken">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#tok0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::tok0"/>
  </nodes>
  <nodes xsi:type="sDocumentStructure:STextualDS">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::salt:/root/doc1#sText1"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::sText1"/>
    <labels xsi:type="saltCore:SFeature" namespace="saltCommon" name="SDATA" value="T::Anöther example for a &apos;Tractaͤtlein&apos;"/>
  </nodes>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.4" target="//@nodes.5">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::7"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.3" target="//@nodes.5">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::8"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::15"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.2" target="//@nodes.5">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::16"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::19"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.1" target="//@nodes.5">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::20"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::21"/>
  </edges>
  <edges xsi:type="sDocumentStructure:STextualRelation" source="//@nodes.0" target="//@nodes.5">
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SSTART" value="N::23"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SEND" value="N::35"/>
  </edges>
  <layers xsi:type="saltCore:SLayer">
    <labels xsi:type="saltCore:SElementId" namespace="salt" name="id" value="T::l0"/>
    <labels xsi:type="saltCore:SFeature" namespace="salt" name="SNAME" value="T::annis"/>
  </layers>
</sDocumentStructure:SDocumentGraph>
